test_conformance/api/test_api_consistency.cpp

```diff
@@ -1151,3 +1179,103 @@ int test_consistency_3d_image_writes(cl_device_id deviceID, cl_context context,
 
     return TEST_PASS;
 }
+
+int test_kernel_local_mem_size(cl_device_id deviceID, cl_context context,
+                               cl_command_queue queue, int num_elements)
+{
+    int error;
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+
+    // Check memory needed to execute empty kernel
+    if (create_single_kernel_helper(context, &program, &kernel, 1, empty_kernel,
+                                    "empty_kernel")
+        != 0)
+    {
+        return -1;
+    }
+
+    cl_ulong kernelLocalUsage = 0;
+    error = clGetKernelWorkGroupInfo(kernel, deviceID, CL_KERNEL_LOCAL_MEM_SIZE,
+                                     sizeof(kernelLocalUsage),
+                                     &kernelLocalUsage, NULL);
+    test_error(error,
+               "clGetKernelWorkGroupInfo for CL_KERNEL_LOCAL_MEM_SIZE failed");
+
+    test_assert_error(kernelLocalUsage > 0, "kernel local mem size failed");
+
+    // Check memory needed to execute empty kernel with __local variable
+    if (create_single_kernel_helper(context, &program, &kernel, 1,
+                                    local_memory_kernel, "local_memory_kernel")
+        != 0)
+    {
+        return -1;
+    }
+
+    kernelLocalUsage = 0;
+    error = clGetKernelWorkGroupInfo(kernel, deviceID, CL_KERNEL_LOCAL_MEM_SIZE,
+                                     sizeof(kernelLocalUsage),
+                                     &kernelLocalUsage, NULL);
+    test_error(error,
+               "clGetKernelWorkGroupInfo for CL_KERNEL_LOCAL_MEM_SIZE failed");
+
+    test_assert_error(kernelLocalUsage >= 10000 * sizeof(cl_float),
+                      "kernel local mem size failed");
+
+    // Check memory needed to execute empty kernel with __local parameter with
+    // setKernelArg
+    if (create_single_kernel_helper(context, &program, &kernel, 1,
+                                    local_param_kernel, "local_param_kernel")
+        != 0)
+    {
+        return -1;
+    }
+
+    size_t elements = 100;
+    size_t sizeToAllocate = elements * sizeof(cl_int);
+    int* localData = (cl_int*)malloc(sizeToAllocate);
```

Comment:
Can we use a smart pointer here rather than malloc (e.g. `make_unique`) so that if the function returns early before the `free(localData)` we don't leak.

Reply:
`localData` is unused.  We may as well remove it entirely.